CMakeLists.txt

```diff
@@ -42,6 +42,7 @@ set_target_properties(vault PROPERTIES
 
 target_include_directories(vault PRIVATE include)
 target_include_directories(vault PRIVATE src)
+target_link_libraries(vault curl)
```

Comment:
I'm not necessarily opposed to linking during the build, but I would like to leave the option not to. In some cases, it's easier and less dangerous to allow the dependency manager or package builder to specify how to link curl to avoid binary compatibility issues. 

Can you put a flag around this to allow users the ability to choose? The default should be `OFF`

Reply:
@jigar23 are you still interested in this?

Reply:
@abedra Sorry for the delay. I have added the option now

Reply:
sorry for the delay @abedra . Added the option now.